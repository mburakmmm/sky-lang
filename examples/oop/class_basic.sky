# Basic OOP test with class, self, and inheritance

class Animal
  function __init__(name: string)
    self.name = name
    self.age = 0
  end

  function speak()
    print("Animal")
    print(self.name)
  end

  function setAge(years: int)
    self.age = years
  end

  function getAge(): int
    return self.age
  end
end

class Dog(Animal)
  function speak()
    print("Woof! I am")
    print(self.name)
  end

  function wagTail()
    print("Wagging tail!")
  end
end

function main
  print("=== Basic Class Test ===")
  let animal = Animal("Generic")
  animal.speak()
  animal.setAge(5)
  print("Age:")
  print(animal.getAge())

  print("=== Inheritance Test ===")
  let dog = Dog("Buddy")
  dog.speak()
  dog.setAge(3)
  print("Dog age:")
  print(dog.getAge())
  dog.wagTail()

  print("=== Done ===")
end

