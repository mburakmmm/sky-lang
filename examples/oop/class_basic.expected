=== Basic Class Test ===
Animal
Generic
Age:
5
=== Inheritance Test ===
Woof! I am
Buddy
Dog age:
3
Wagging tail!
=== Done ===

